Cmdline/Action/List.cs

```diff
@@ -65,26 +65,49 @@ public int RunCommand(CKAN.KSP ksp, object raw_options)
                     else if (current_version is DllVersion)
                     {
                         // Autodetected dll
-                        bullet = "-";
+                        bullet = "A";
                     }
                     else
                     {
                         try
                         {
                             // Check if upgrades are available, and show appropriately.
                             CkanModule latest = registry.LatestAvailable(mod.Key, ksp.VersionCriteria());
-
+                            CkanModule current = registry.GetInstalledVersion(mod.Key);
+                                
                             log.InfoFormat("Latest {0} is {1}", mod.Key, latest);
 
                             if (latest == null)
                             {
                                 // Not compatible!
                                 bullet = "X";
+                                //Check if mod is replaceable
+                                if ( current.replaced_by != null)
+                                {
+                                    CkanModule replacement = registry.LatestAvailable(current.replaced_by.name, ksp.VersionCriteria());
```

Comment:
`LatestAvailable` throws `ModuleNotFoundKraken` if the passed mod is not compatible with the active version, which will be caught at line 118 and trigger an error message. However, that error message will be misleading since it says the original replaced mod is not in the registry.

I think we need `try`/`catch` blocks around these `LatestAvailable` calls.

(Same comment for the identical block below)

Reply:
What happens when you nest try / catch blocks?

Reply:
It works as you'd expect, the inner `catch` handles exceptions inside the inner `try` block, and the outer `catch` handles exceptions in the outer `try` block. If the inner `catch` doesn't handle an exception in the inner block or or re-throws it, it gets passed to the outer `catch`.